packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/index.tsx

```diff
@@ -156,35 +143,83 @@ const ConfirmSignupCode = (_: RouteComponentProps) => {
-          REACT_ENTRYPOINT
+        // to match parity with content-server event, viewName is NOT included
+        logViewEvent(`flow`, 'newsletter.subscribed', REACT_ENTRYPOINT);
+      }
+
+      if (isSyncDesktopIntegration(integration)) {
+        // Tell FF the account is verified with webchannel event, FXA-8287
+        // hardNavigateToContentServer('/connect_another_device')
+      } else if (isOAuthIntegration(integration)) {
+        // Check to see if the relier wants TOTP.
+        // Newly created accounts wouldn't have this so lets redirect them to signin.
+        // Certain reliers may require users to set up 2FA / TOTP
+        // before they can be redirected back to the RP.
+        // Notes in content-server indicate that a message should be displayed on the signin page
+        // to explain why totp setup is required, but this does not currently
+        // appear to be implemented.
+
+        // TODO enable wantsTwoStepAuthentication check
+        // Currently, does not correctly check for acr_values param
+        // and is returned as false even if search params contain acr_values=AAL2 (or higher)
+
+        // Params are included to eventually allow for redirect to RP after 2FA setup
+        if (integration.wantsTwoStepAuthentication()) {
+          hardNavigateToContentServer(`oauth/signin${window.location.search}`);
+          return;
+        } else {
+          const { redirect } = await finishOAuthFlowHandler(
+            integration.data.uid,
+            sessionToken,
+            keyFetchToken,
+            unwrapBKey
+          );
+
+          // Navigate to relying party
+          hardNavigate(redirect);
+          return;
+        }
+      } else {
+        // TODO: Check if we ever want to show 'signup_confirmed' (Ready view)
+        // Backbone had a base navigation behaviour navigating there
+        alertBar.success(
+          ftlMsgResolver.getMsg(
+            'confirm-signup-code-success-alert',
+            'Account confirmed successfully'
+          )
         );
+        navigate('/settings', { replace: true });
       }
 
-      // FOLLOW-UP: Broker not yet implemented
-      // The broker handles navigation behaviour that varies depending on the relier
-      // and may include web channel notifications to ensure the verification is propagated
-      // to other tabs
-      // await broker.invokeBrokerMethod('afterSignUpConfirmationPoll', account);
-      // This may be taken care of with^ but we need to send a web channel message
-      // to FF to tell it the account was verified. Can be done with FXA-8287 or follow up
-      alertSuccessAndGoForward();
-    } catch (e) {
-      const localizedErrorMessage = ftlMsgResolver.getMsg(
-        composeAuthUiErrorTranslationId(e),
-        e.message
-      );
-      // If the error is one of the three indicated types, display error in input tooltip
+      // TODO: Do we need a ticket for unpersistVerificationData, or remove
```

Comment:
We should be able to remove all the TODOs around this.

We used to store "verification info" in `localStorage` so the same auth-broker was used if the user verifies in a different browser. See [this explainer](https://docs.google.com/document/d/1K4AD69QgfOCZwFLp7rUcMOkOTslbLCh7jjSdR9zpAkk/edit#heading=h.kkt4eylho93t) (thanks past me).

Reply:
Nice, I'll remove references to `unpersistVerificationData` in fxa-settings